core/templates/dev/head/pages/profile/profile.html

```diff
@@ -114,7 +114,7 @@ <h2 class="oppia-profile-username-small-screen">
 
     <md-card class="oppia-profile-content-card">
       <md-content class="oppia-profile-portfolio-container">
-        <exploration-summary-tile ng-repeat="expl in getExplorationsToDisplay()"
+        <exploration-summary-tile ng-repeat="expl in getExplorationsToDisplay() track by $index"
                                   exploration-id="expl.id"
```

Comment:
What? Why did you need to make changes here?

Instead of using ng-class with a dict, Sean and I suggested you to use ng-repeat(specifically `track by $index`) so that you can apply it to the `<i>button<i>` down below, not here. Does that make sense?

Reply:
Sorry..!
I don't understand you.
**ng-class** and **ng-repeat** both are different right..!!
how can we use **ng-repeat** instead of **ng-class**.

based on this [doc](https://docs.angularjs.org/api/ng/directive/ngRepeat#tracking-and-duplicates) i wrote this line 
> ng-repeat="expl in getExplorationsToDisplay() track by $index"

Reply:
I mean its a suggestion. If you deem it too hard, you can find more straightforward alternative. How about ng-disable/ng-if?

Reply:
Wait, hang on. I think "track by $index" here is correct, it fixes the console error right?

Reply:
I was just talking about the `ng-class` usage. I am fine with the changes here on 2nd thought. Thanks!

Reply:
There is no need of **track by $index** here.
The actual problem is with below statement.
> $scope.currentPageNumber + 1) * $scope.PAGE_SIZE

I have added **track by $index**  based on these [docs](https://docs.angularjs.org/api/ng/directive/ngRepeat#tracking-and-duplicates).It says ****track by $index** will saves time by not creating **dom element** every time.

Thanks.

Reply:
Hang on -- in that case, which of the changes you made fixed the console error you mentioned earlier?

Reply:
<pre>` 
      $scope.goToNextPage = function() {
        if ($scope.currentPageNumber  * $scope.PAGE_SIZE >= (
          data.edited_exp_summary_dicts.length)) {
          $log.error('Error: Cannot increment page');
        } else {
          $scope.currentPageNumber++;
          $scope.startingExplorationNumber = (
            $scope.currentPageNumber * $scope.PAGE_SIZE + 1
          );
          $scope.endingExplorationNumber = (
            Math.min($scope.numUserPortfolioExplorations,
              ($scope.currentPageNumber + 1) * $scope.PAGE_SIZE)
          );
        }
      };
`
</pre>
**to**
<pre>
`
      $scope.goToNextPage = function() {
        if (($scope.currentPageNumber <b><u>+ 1)</u></b> * $scope.PAGE_SIZE >= (
          data.edited_exp_summary_dicts.length)) {
          $log.error('Error: Cannot increment page');
        } else {
          $scope.currentPageNumber++;
          $scope.startingExplorationNumber = (
            $scope.currentPageNumber * $scope.PAGE_SIZE + 1
          );
          $scope.endingExplorationNumber = (
            Math.min($scope.numUserPortfolioExplorations,
              ($scope.currentPageNumber + 1) * $scope.PAGE_SIZE)
          );
        }
      };
`
</pre>
This is the change i have done.which removes console error.!
Thanks.

Reply:
Oh! I didn't expect that at all, interesting. Thanks for explaining.